### PR TITLE
ISPN-4919 Configuration templates

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-4919
https://issues.jboss.org/browse/ISPN-5656

Implement configuration templates.
On the server this was done by separating the model of caches and configurations.

### CODE REVIEW

**Comment by tristantarrant** [comment]
Leaving this @ preview for the moment as I have known failures due to the huge management model refactor. As soon as I've ironed these out, I'll upgrade to "ready to review". Nothing stops you from reviewing this anyway ;)

**Comment by tristantarrant** [comment]
This is now ready

**Comment by tristantarrant** [comment]
The jcache remote code needs to be updated. Working on it

**Comment by tristantarrant** [comment]
That should do it. CI do your thang !

**Comment by tristantarrant** [comment]
And the testsuite is green !

File: core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java [hunk 1]
```diff
@@ -208,6 +210,38 @@
...
+    * @return null if no configuration exists as per rules set above, otherwise
+    *         returns a cache instance identified by cacheName
+    */
+   <K, V> Cache<K, V> getCache(String cacheName, String configurationName);
+
+   /**
+    * Retrieves a named cache from the system in the same way that {@link
+    * #getCache(String)} does except that if offers the possibility to specify
+    * a specific configuration template. Multiple caches can be created using
+    * the same configuration. Tihs method also offers the possibility for the
```
Comment (galderz) [inline core/src/main/java/org/infinispan/manager/EmbeddedCacheManager.java]: Minor spelling error

File: server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheAdd.java [hunk 2]
```diff
@@ -275,15 +139,15 @@ void removeRuntimeServices(OperationContext context, ModelNode operation, ModelN
         final String containerName = containerAddress.getLastElement().getValue() ;
 
         // remove all services started by CacheAdd, in reverse order
+
+        // FIXME restore JNDI removal
```
Comment (galderz) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/CacheAdd.java]: Is there a JIRA tracking this?

**Comment by galderz** [comment]
Minor comments, so integrated.

**Pull request closed without merging**